{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sequencing G001"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pathlib import Path\n",
    "import glob\n",
    "\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "\n",
    "import my_functions as fun\n",
    "pd.set_option('display.max_columns', 25)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load SNP overview"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this dataframe you can find which genes of the WT are mutated in G001, and e.g. how severe the mutation is."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# open .txt file and read text \n",
    "f = open(fun.get_file_path('P1_bowtie2_20201011.variants-final.Refseq.snpEff_genes.txt'), 'r')\n",
    "content = f.read()\n",
    "f.close()\n",
    "\n",
    "# split text into lines and tabs\n",
    "output = content.split('\\n')\n",
    "output = [n.split('\\t') for n in output]\n",
    "\n",
    "# discard first row, does not contain data\n",
    "del output[0]\n",
    "\n",
    "# create dataframe and identify first row as column names\n",
    "df_genes = pd.DataFrame(output)\n",
    "df_genes.columns = df_genes.iloc[0]\n",
    "df_genes = df_genes.drop(0)\n",
    "df_genes = df_genes[:-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#GeneName</th>\n",
       "      <th>GeneId</th>\n",
       "      <th>TranscriptId</th>\n",
       "      <th>BioType</th>\n",
       "      <th>variants_impact_HIGH</th>\n",
       "      <th>variants_impact_LOW</th>\n",
       "      <th>variants_impact_MODERATE</th>\n",
       "      <th>variants_impact_MODIFIER</th>\n",
       "      <th>variants_effect_conservative_inframe_deletion</th>\n",
       "      <th>variants_effect_conservative_inframe_insertion</th>\n",
       "      <th>variants_effect_disruptive_inframe_deletion</th>\n",
       "      <th>variants_effect_downstream_gene_variant</th>\n",
       "      <th>variants_effect_frameshift_variant</th>\n",
       "      <th>variants_effect_missense_variant</th>\n",
       "      <th>variants_effect_splice_region_variant</th>\n",
       "      <th>variants_effect_stop_gained</th>\n",
       "      <th>variants_effect_stop_lost</th>\n",
       "      <th>variants_effect_synonymous_variant</th>\n",
       "      <th>variants_effect_upstream_gene_variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CGZ69_RS00005</td>\n",
       "      <td>CGZ69_RS00005</td>\n",
       "      <td>CGZ69_RS00005</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CGZ69_RS00010</td>\n",
       "      <td>CGZ69_RS00010</td>\n",
       "      <td>CGZ69_RS00010</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CGZ69_RS00015</td>\n",
       "      <td>CGZ69_RS00015</td>\n",
       "      <td>CGZ69_RS00015</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CGZ69_RS00020</td>\n",
       "      <td>CGZ69_RS00020</td>\n",
       "      <td>CGZ69_RS00020</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CGZ69_RS00030</td>\n",
       "      <td>CGZ69_RS00030</td>\n",
       "      <td>CGZ69_RS00030</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5621</th>\n",
       "      <td>yidC</td>\n",
       "      <td>CGZ69_RS13080</td>\n",
       "      <td>CGZ69_RS13080</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5622</th>\n",
       "      <td>yidC</td>\n",
       "      <td>CGZ69_RS18120</td>\n",
       "      <td>CGZ69_RS18120</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5623</th>\n",
       "      <td>yidD</td>\n",
       "      <td>CGZ69_RS18125</td>\n",
       "      <td>CGZ69_RS18125</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5624</th>\n",
       "      <td>zapE</td>\n",
       "      <td>CGZ69_RS29290</td>\n",
       "      <td>CGZ69_RS29290</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5625</th>\n",
       "      <td>zwf</td>\n",
       "      <td>CGZ69_RS34785</td>\n",
       "      <td>CGZ69_RS34785</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5625 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "0         #GeneName         GeneId   TranscriptId         BioType  \\\n",
       "1     CGZ69_RS00005  CGZ69_RS00005  CGZ69_RS00005  protein_coding   \n",
       "2     CGZ69_RS00010  CGZ69_RS00010  CGZ69_RS00010  protein_coding   \n",
       "3     CGZ69_RS00015  CGZ69_RS00015  CGZ69_RS00015  protein_coding   \n",
       "4     CGZ69_RS00020  CGZ69_RS00020  CGZ69_RS00020  protein_coding   \n",
       "5     CGZ69_RS00030  CGZ69_RS00030  CGZ69_RS00030  protein_coding   \n",
       "...             ...            ...            ...             ...   \n",
       "5621           yidC  CGZ69_RS13080  CGZ69_RS13080  protein_coding   \n",
       "5622           yidC  CGZ69_RS18120  CGZ69_RS18120  protein_coding   \n",
       "5623           yidD  CGZ69_RS18125  CGZ69_RS18125  protein_coding   \n",
       "5624           zapE  CGZ69_RS29290  CGZ69_RS29290  protein_coding   \n",
       "5625            zwf  CGZ69_RS34785  CGZ69_RS34785  protein_coding   \n",
       "\n",
       "0    variants_impact_HIGH variants_impact_LOW variants_impact_MODERATE  \\\n",
       "1                       0                   0                        0   \n",
       "2                       0                   0                        0   \n",
       "3                       0                   0                        0   \n",
       "4                       0                   0                        0   \n",
       "5                       0                   0                        0   \n",
       "...                   ...                 ...                      ...   \n",
       "5621                    0                   0                        0   \n",
       "5622                    0                   0                        0   \n",
       "5623                    0                   0                        0   \n",
       "5624                    0                   0                        1   \n",
       "5625                    0                   0                        0   \n",
       "\n",
       "0    variants_impact_MODIFIER variants_effect_conservative_inframe_deletion  \\\n",
       "1                           1                                             0   \n",
       "2                           1                                             0   \n",
       "3                           1                                             0   \n",
       "4                           1                                             0   \n",
       "5                           1                                             0   \n",
       "...                       ...                                           ...   \n",
       "5621                        1                                             0   \n",
       "5622                        2                                             0   \n",
       "5623                        1                                             0   \n",
       "5624                        7                                             0   \n",
       "5625                        4                                             0   \n",
       "\n",
       "0    variants_effect_conservative_inframe_insertion  \\\n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "5                                                 0   \n",
       "...                                             ...   \n",
       "5621                                              0   \n",
       "5622                                              0   \n",
       "5623                                              0   \n",
       "5624                                              0   \n",
       "5625                                              0   \n",
       "\n",
       "0    variants_effect_disruptive_inframe_deletion  \\\n",
       "1                                              0   \n",
       "2                                              0   \n",
       "3                                              0   \n",
       "4                                              0   \n",
       "5                                              0   \n",
       "...                                          ...   \n",
       "5621                                           0   \n",
       "5622                                           0   \n",
       "5623                                           0   \n",
       "5624                                           0   \n",
       "5625                                           0   \n",
       "\n",
       "0    variants_effect_downstream_gene_variant  \\\n",
       "1                                          1   \n",
       "2                                          0   \n",
       "3                                          0   \n",
       "4                                          1   \n",
       "5                                          0   \n",
       "...                                      ...   \n",
       "5621                                       1   \n",
       "5622                                       2   \n",
       "5623                                       1   \n",
       "5624                                       5   \n",
       "5625                                       2   \n",
       "\n",
       "0    variants_effect_frameshift_variant variants_effect_missense_variant  \\\n",
       "1                                     0                                0   \n",
       "2                                     0                                0   \n",
       "3                                     0                                0   \n",
       "4                                     0                                0   \n",
       "5                                     0                                0   \n",
       "...                                 ...                              ...   \n",
       "5621                                  0                                0   \n",
       "5622                                  0                                0   \n",
       "5623                                  0                                0   \n",
       "5624                                  0                                1   \n",
       "5625                                  0                                0   \n",
       "\n",
       "0    variants_effect_splice_region_variant variants_effect_stop_gained  \\\n",
       "1                                        0                           0   \n",
       "2                                        0                           0   \n",
       "3                                        0                           0   \n",
       "4                                        0                           0   \n",
       "5                                        0                           0   \n",
       "...                                    ...                         ...   \n",
       "5621                                     0                           0   \n",
       "5622                                     0                           0   \n",
       "5623                                     0                           0   \n",
       "5624                                     0                           0   \n",
       "5625                                     0                           0   \n",
       "\n",
       "0    variants_effect_stop_lost variants_effect_synonymous_variant  \\\n",
       "1                            0                                  0   \n",
       "2                            0                                  0   \n",
       "3                            0                                  0   \n",
       "4                            0                                  0   \n",
       "5                            0                                  0   \n",
       "...                        ...                                ...   \n",
       "5621                         0                                  0   \n",
       "5622                         0                                  0   \n",
       "5623                         0                                  0   \n",
       "5624                         0                                  0   \n",
       "5625                         0                                  0   \n",
       "\n",
       "0    variants_effect_upstream_gene_variant  \n",
       "1                                        0  \n",
       "2                                        1  \n",
       "3                                        1  \n",
       "4                                        0  \n",
       "5                                        1  \n",
       "...                                    ...  \n",
       "5621                                     0  \n",
       "5622                                     0  \n",
       "5623                                     0  \n",
       "5624                                     2  \n",
       "5625                                     2  \n",
       "\n",
       "[5625 rows x 19 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_genes\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# modify the dataframe with all genes, to be able to search more easily\n",
    "df_genes[\"GeneId\"] = df_genes[\"GeneId\"].str.replace(\"CGZ69_RS\", \"\")\n",
    "df_genes[\"GeneId\"] = pd.to_numeric(df_genes[\"GeneId\"])\n",
    "df_genes[\"variants_impact_HIGH\"] = pd.to_numeric(df_genes[\"variants_impact_HIGH\"])\n",
    "df_genes[\"variants_impact_MODERATE\"] = pd.to_numeric(df_genes[\"variants_impact_MODERATE\"])\n",
    "df_genes[\"variants_impact_LOW\"] = pd.to_numeric(df_genes[\"variants_impact_LOW\"])\n",
    "df_genes[\"variants_impact_MODIFIER\"] = pd.to_numeric(df_genes[\"variants_impact_MODIFIER\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#GeneName</th>\n",
       "      <th>GeneId</th>\n",
       "      <th>TranscriptId</th>\n",
       "      <th>BioType</th>\n",
       "      <th>variants_impact_HIGH</th>\n",
       "      <th>variants_impact_LOW</th>\n",
       "      <th>variants_impact_MODERATE</th>\n",
       "      <th>variants_impact_MODIFIER</th>\n",
       "      <th>variants_effect_conservative_inframe_deletion</th>\n",
       "      <th>variants_effect_conservative_inframe_insertion</th>\n",
       "      <th>variants_effect_disruptive_inframe_deletion</th>\n",
       "      <th>variants_effect_downstream_gene_variant</th>\n",
       "      <th>variants_effect_frameshift_variant</th>\n",
       "      <th>variants_effect_missense_variant</th>\n",
       "      <th>variants_effect_splice_region_variant</th>\n",
       "      <th>variants_effect_stop_gained</th>\n",
       "      <th>variants_effect_stop_lost</th>\n",
       "      <th>variants_effect_synonymous_variant</th>\n",
       "      <th>variants_effect_upstream_gene_variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CGZ69_RS00005</td>\n",
       "      <td>5</td>\n",
       "      <td>CGZ69_RS00005</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CGZ69_RS00010</td>\n",
       "      <td>10</td>\n",
       "      <td>CGZ69_RS00010</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CGZ69_RS00015</td>\n",
       "      <td>15</td>\n",
       "      <td>CGZ69_RS00015</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CGZ69_RS00020</td>\n",
       "      <td>20</td>\n",
       "      <td>CGZ69_RS00020</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CGZ69_RS00030</td>\n",
       "      <td>30</td>\n",
       "      <td>CGZ69_RS00030</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "0      #GeneName  GeneId   TranscriptId         BioType  variants_impact_HIGH  \\\n",
       "1  CGZ69_RS00005       5  CGZ69_RS00005  protein_coding                     0   \n",
       "2  CGZ69_RS00010      10  CGZ69_RS00010  protein_coding                     0   \n",
       "3  CGZ69_RS00015      15  CGZ69_RS00015  protein_coding                     0   \n",
       "4  CGZ69_RS00020      20  CGZ69_RS00020  protein_coding                     0   \n",
       "5  CGZ69_RS00030      30  CGZ69_RS00030  protein_coding                     0   \n",
       "\n",
       "0  variants_impact_LOW  variants_impact_MODERATE  variants_impact_MODIFIER  \\\n",
       "1                    0                         0                         1   \n",
       "2                    0                         0                         1   \n",
       "3                    0                         0                         1   \n",
       "4                    0                         0                         1   \n",
       "5                    0                         0                         1   \n",
       "\n",
       "0 variants_effect_conservative_inframe_deletion  \\\n",
       "1                                             0   \n",
       "2                                             0   \n",
       "3                                             0   \n",
       "4                                             0   \n",
       "5                                             0   \n",
       "\n",
       "0 variants_effect_conservative_inframe_insertion  \\\n",
       "1                                              0   \n",
       "2                                              0   \n",
       "3                                              0   \n",
       "4                                              0   \n",
       "5                                              0   \n",
       "\n",
       "0 variants_effect_disruptive_inframe_deletion  \\\n",
       "1                                           0   \n",
       "2                                           0   \n",
       "3                                           0   \n",
       "4                                           0   \n",
       "5                                           0   \n",
       "\n",
       "0 variants_effect_downstream_gene_variant variants_effect_frameshift_variant  \\\n",
       "1                                       1                                  0   \n",
       "2                                       0                                  0   \n",
       "3                                       0                                  0   \n",
       "4                                       1                                  0   \n",
       "5                                       0                                  0   \n",
       "\n",
       "0 variants_effect_missense_variant variants_effect_splice_region_variant  \\\n",
       "1                                0                                     0   \n",
       "2                                0                                     0   \n",
       "3                                0                                     0   \n",
       "4                                0                                     0   \n",
       "5                                0                                     0   \n",
       "\n",
       "0 variants_effect_stop_gained variants_effect_stop_lost  \\\n",
       "1                           0                         0   \n",
       "2                           0                         0   \n",
       "3                           0                         0   \n",
       "4                           0                         0   \n",
       "5                           0                         0   \n",
       "\n",
       "0 variants_effect_synonymous_variant variants_effect_upstream_gene_variant  \n",
       "1                                  0                                     0  \n",
       "2                                  0                                     1  \n",
       "3                                  0                                     1  \n",
       "4                                  0                                     0  \n",
       "5                                  0                                     1  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_genes.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load SNP annotation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this dataframe you can look up what type of mutation has occured in each gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# open the .tsv file and read into a dataframe\n",
    "df_ann = pd.read_csv(fun.get_file_path(\"P1_bowtie2_20201011.variants-final.Refseq.ann.tsv\"), sep=\"\\t\")\n",
    "df_ann = df_ann.drop(labels='Unnamed: 5', axis=1)  # remove last column that does not contain data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create dataframe of all genes that are affected by each mutation \n",
    "df_ann_headers = df_ann['ANN'].str.split(', ', expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# obtain the annotation column IDs from the .vcf file. \n",
    "with open(fun.get_file_path(\"P1_bowtie2_20201011.variants-final.Refseq.ann.vcf\")) as file:\n",
    "    while line := file.readline(): \n",
    "        if line.startswith(\"##INFO=<ID=ANN\"):  # find the row with annotation IDs\n",
    "            ann_ids_raw = line\n",
    "            break\n",
    "\n",
    "ann_ids = ann_ids_raw.split(\"'\")[1].split(' | ')  # format annotation IDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# expand data of first gene that is affected by the mutation\n",
    "df_ann[ann_ids] = df_ann_headers[0].str.split('|', expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# modify the dataframe with all genes, to be able to search more easily\n",
    "df_ann[\"Distance\"] = pd.to_numeric(df_ann[\"Distance\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>ANN</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Annotation_Impact</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Feature_Type</th>\n",
       "      <th>Feature_ID</th>\n",
       "      <th>Transcript_BioType</th>\n",
       "      <th>Rank</th>\n",
       "      <th>HGVS.c</th>\n",
       "      <th>HGVS.p</th>\n",
       "      <th>cDNA.pos / cDNA.length</th>\n",
       "      <th>CDS.pos / CDS.length</th>\n",
       "      <th>AA.pos / AA.length</th>\n",
       "      <th>Distance</th>\n",
       "      <th>ERRORS / WARNINGS / INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>AG</td>\n",
       "      <td>AGAGCGGGTAAGAAG</td>\n",
       "      <td>96.0</td>\n",
       "      <td>AGAGCGGGTAAGAAG|upstream_gene_variant|MODIFIER...</td>\n",
       "      <td>AGAGCGGGTAAGAAG</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>CGZ69_RS00010</td>\n",
       "      <td>CGZ69_RS00010</td>\n",
       "      <td>transcript</td>\n",
       "      <td>CGZ69_RS00010</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>c.-1277_-1276insAGCGGGTAAGAAG</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1276.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>48701</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>225.0</td>\n",
       "      <td>T|missense_variant|MODERATE|CGZ69_RS00225|CGZ6...</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>CGZ69_RS00225</td>\n",
       "      <td>CGZ69_RS00225</td>\n",
       "      <td>transcript</td>\n",
       "      <td>CGZ69_RS00225</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td>c.371G&gt;A</td>\n",
       "      <td>p.Arg124Lys</td>\n",
       "      <td>371/748</td>\n",
       "      <td>371/748</td>\n",
       "      <td>124/248</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WARNING_TRANSCRIPT_INCOMPLETE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>56021</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>225.0</td>\n",
       "      <td>T|missense_variant|MODERATE|CGZ69_RS00245|CGZ6...</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>CGZ69_RS00245</td>\n",
       "      <td>CGZ69_RS00245</td>\n",
       "      <td>transcript</td>\n",
       "      <td>CGZ69_RS00245</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td>c.1798G&gt;A</td>\n",
       "      <td>p.Glu600Lys</td>\n",
       "      <td>1798/6354</td>\n",
       "      <td>1798/6354</td>\n",
       "      <td>600/2117</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>83126</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>225.0</td>\n",
       "      <td>A|missense_variant|MODERATE|CGZ69_RS36435|CGZ6...</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>CGZ69_RS36435</td>\n",
       "      <td>CGZ69_RS36435</td>\n",
       "      <td>transcript</td>\n",
       "      <td>CGZ69_RS36435</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td>c.13G&gt;A</td>\n",
       "      <td>p.Ala5Thr</td>\n",
       "      <td>13/153</td>\n",
       "      <td>13/153</td>\n",
       "      <td>5/50</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>100949</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>225.0</td>\n",
       "      <td>G|missense_variant|MODERATE|cas7e|CGZ69_RS0044...</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>cas7e</td>\n",
       "      <td>CGZ69_RS00445</td>\n",
       "      <td>transcript</td>\n",
       "      <td>CGZ69_RS00445</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td>c.1130G&gt;C</td>\n",
       "      <td>p.Gly377Ala</td>\n",
       "      <td>1130/1155</td>\n",
       "      <td>1130/1155</td>\n",
       "      <td>377/384</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WARNING_TRANSCRIPT_NO_START_CODON</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      POS REF              ALT   QUAL  \\\n",
       "0       4  AG  AGAGCGGGTAAGAAG   96.0   \n",
       "1   48701   C                T  225.0   \n",
       "2   56021   C                T  225.0   \n",
       "3   83126   G                A  225.0   \n",
       "4  100949   C                G  225.0   \n",
       "\n",
       "                                                 ANN           Allele  \\\n",
       "0  AGAGCGGGTAAGAAG|upstream_gene_variant|MODIFIER...  AGAGCGGGTAAGAAG   \n",
       "1  T|missense_variant|MODERATE|CGZ69_RS00225|CGZ6...                T   \n",
       "2  T|missense_variant|MODERATE|CGZ69_RS00245|CGZ6...                T   \n",
       "3  A|missense_variant|MODERATE|CGZ69_RS36435|CGZ6...                A   \n",
       "4  G|missense_variant|MODERATE|cas7e|CGZ69_RS0044...                G   \n",
       "\n",
       "              Annotation Annotation_Impact      Gene_Name        Gene_ID  \\\n",
       "0  upstream_gene_variant          MODIFIER  CGZ69_RS00010  CGZ69_RS00010   \n",
       "1       missense_variant          MODERATE  CGZ69_RS00225  CGZ69_RS00225   \n",
       "2       missense_variant          MODERATE  CGZ69_RS00245  CGZ69_RS00245   \n",
       "3       missense_variant          MODERATE  CGZ69_RS36435  CGZ69_RS36435   \n",
       "4       missense_variant          MODERATE          cas7e  CGZ69_RS00445   \n",
       "\n",
       "  Feature_Type     Feature_ID Transcript_BioType Rank  \\\n",
       "0   transcript  CGZ69_RS00010     protein_coding        \n",
       "1   transcript  CGZ69_RS00225     protein_coding  1/1   \n",
       "2   transcript  CGZ69_RS00245     protein_coding  1/1   \n",
       "3   transcript  CGZ69_RS36435     protein_coding  1/1   \n",
       "4   transcript  CGZ69_RS00445     protein_coding  1/1   \n",
       "\n",
       "                          HGVS.c       HGVS.p cDNA.pos / cDNA.length  \\\n",
       "0  c.-1277_-1276insAGCGGGTAAGAAG                                       \n",
       "1                       c.371G>A  p.Arg124Lys                371/748   \n",
       "2                      c.1798G>A  p.Glu600Lys              1798/6354   \n",
       "3                        c.13G>A    p.Ala5Thr                 13/153   \n",
       "4                      c.1130G>C  p.Gly377Ala              1130/1155   \n",
       "\n",
       "  CDS.pos / CDS.length AA.pos / AA.length  Distance  \\\n",
       "0                                            1276.0   \n",
       "1              371/748            124/248       NaN   \n",
       "2            1798/6354           600/2117       NaN   \n",
       "3               13/153               5/50       NaN   \n",
       "4            1130/1155            377/384       NaN   \n",
       "\n",
       "            ERRORS / WARNINGS / INFO  \n",
       "0                                     \n",
       "1      WARNING_TRANSCRIPT_INCOMPLETE  \n",
       "2                                     \n",
       "3                                     \n",
       "4  WARNING_TRANSCRIPT_NO_START_CODON  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ann.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## SCO annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# open .txt file and read text \n",
    "f = open(fun.get_file_path('NZ_CP022438_S.peucetius_subsp.caesius_ATCC_27952_chromosome_proteins_toSCO.tsv_corr_e0.001_cover60.tsv'), 'r')\n",
    "content = f.read()\n",
    "f.close()\n",
    "\n",
    "# split text into lines and tabs\n",
    "output = content.splitlines()\n",
    "output = [n.split('\\t') for n in output]\n",
    "\n",
    "# create dataframe and identify first row as column names\n",
    "df_SCO = pd.DataFrame(output)\n",
    "df_SCO.columns = df_SCO.iloc[0]\n",
    "df_SCO = df_SCO.drop(0)\n",
    "df_SCO = df_SCO[:-1]\n",
    "df_SCO[[\"SCO\", \"SCO_function\"]] = df_SCO[\"SCO\"].str.split(\" \", 1, expand=True)\n",
    "df_SCO[[\"S.peucetius\", \"SPE_code\"]] = df_SCO[\"S.peucetius\"].str.split(\" \", 1, expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S.peucetius</th>\n",
       "      <th>SCO</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>SCO_function</th>\n",
       "      <th>SPE_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CGZ69_RS00005</td>\n",
       "      <td>NotFound</td>\n",
       "      <td>0%</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CGZ69_RS00010</td>\n",
       "      <td>SCO1013</td>\n",
       "      <td>95%</td>\n",
       "      <td>mut-like protein</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CGZ69_RS00015</td>\n",
       "      <td>NotFound</td>\n",
       "      <td>0%</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CGZ69_RS00020</td>\n",
       "      <td>NotFound</td>\n",
       "      <td>0%</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CGZ69_RS00035</td>\n",
       "      <td>NotFound</td>\n",
       "      <td>0%</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6398</th>\n",
       "      <td>CGZ69_RS36200</td>\n",
       "      <td>NotFound</td>\n",
       "      <td>0%</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6399</th>\n",
       "      <td>CGZ69_RS36205</td>\n",
       "      <td>SCO5327</td>\n",
       "      <td>99%</td>\n",
       "      <td>transposase</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6400</th>\n",
       "      <td>CGZ69_RS36210</td>\n",
       "      <td>SCO0768</td>\n",
       "      <td>106%</td>\n",
       "      <td>lipoprotein</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6401</th>\n",
       "      <td>CGZ69_RS36220</td>\n",
       "      <td>NotFound</td>\n",
       "      <td>0%</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6402</th>\n",
       "      <td>CGZ69_RS37050</td>\n",
       "      <td>NotFound</td>\n",
       "      <td>0%</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6402 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "0       S.peucetius       SCO Coverage      SCO_function SPE_code\n",
       "1     CGZ69_RS00005  NotFound       0%              None     None\n",
       "2     CGZ69_RS00010   SCO1013      95%  mut-like protein     None\n",
       "3     CGZ69_RS00015  NotFound       0%              None     None\n",
       "4     CGZ69_RS00020  NotFound       0%              None     None\n",
       "5     CGZ69_RS00035  NotFound       0%              None     None\n",
       "...             ...       ...      ...               ...      ...\n",
       "6398  CGZ69_RS36200  NotFound       0%              None     None\n",
       "6399  CGZ69_RS36205   SCO5327      99%       transposase     None\n",
       "6400  CGZ69_RS36210   SCO0768     106%       lipoprotein     None\n",
       "6401  CGZ69_RS36220  NotFound       0%              None     None\n",
       "6402  CGZ69_RS37050  NotFound       0%              None     None\n",
       "\n",
       "[6402 rows x 5 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_SCO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_SCO = df_SCO.sort_values(\"S.peucetius\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_SCO.to_csv(\"df_SCO.csv\", index=False, sep = ';')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a dictionary of SCO numbers\n",
    "dict_SCO = {gene_code: \"pseudo_gene\" for gene_code in df_genes[\"TranscriptId\"]}\n",
    "\n",
    "for gene_code in df_SCO[\"S.peucetius\"]:\n",
    "    dict_SCO[gene_code] = df_SCO[df_SCO[\"S.peucetius\"] == gene_code][\"SCO\"].iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add SCO number to df_ann and df_genes\n",
    "df_ann[\"SCO\"] = df_ann[\"Gene_ID\"].apply(lambda x: dict_SCO[x])\n",
    "df_genes[\"SCO\"] = df_genes[\"TranscriptId\"].apply(lambda x: dict_SCO[x])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_genes.to_csv(\"df_genes.csv\", index=False, sep = ';')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ann.to_csv(\"df_ann.csv\", index=False, sep = ';')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Add AntiSMASH results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Collect filenames of all gbk files with antismash clusters in a list\n",
    "list_files = []\n",
    "\n",
    "for path in Path('C:\\\\Users\\\\mandy\\\\Documents\\\\Programming\\\\G001_sequencing\\\\input\\\\').rglob('*region*'):\n",
    "    list_files.append(path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Collect all antismash clusters in a list\n",
    "list_clusters = [filename.name.replace(\"NZ_CP022438.1.\", \"\").replace(\".gbk\", \"\") for filename in list_files]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Extract all the genes (CDS actually) from each cluster and collect in dictionary\n",
    "\n",
    "# initialize dictionary to store all genes for each cluster\n",
    "dict_antismash = {gene_code: \"\" for gene_code in df_genes[\"TranscriptId\"]}\n",
    "cluster_product = \"?\"\n",
    "\n",
    "# TODO: change to enumerate\n",
    "for num in range(len(list_files)):\n",
    "    \n",
    "    # open the file with BioPython\n",
    "    record = SeqIO.read(list_files[num], \"genbank\")\n",
    "\n",
    "    # initialize a list to collect all genes in the cluster\n",
    "    list_genes = []\n",
    "\n",
    "    for feat_num in range(len(record.features)):\n",
    "        \n",
    "        feat = record.features[feat_num]\n",
    "        \n",
    "        if feat.type == \"cand_cluster\":\n",
    "            cluster_product = feat.qualifiers[\"product\"][0]\n",
    "            break\n",
    "                \n",
    "    for feat_num in range(len(record.features)):\n",
    "        \n",
    "        # collect all gene names in current record in the current list\n",
    "        feat = record.features[feat_num]\n",
    "\n",
    "        if feat.type == \"CDS\":\n",
    "            gene = feat.qualifiers[\"locus_tag\"][0]\n",
    "            list_genes.append(gene)\n",
    "            dict_antismash[gene] = list_clusters[num] + ', ' + cluster_product"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'allorf_3591657_3591773'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp/ipykernel_2408/2445967785.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[1;32mdel\u001b[0m \u001b[0mdict_antismash\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'allorf_3591657_3591773'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mKeyError\u001b[0m: 'allorf_3591657_3591773'"
     ]
    }
   ],
   "source": [
    "del dict_antismash['allorf_3591657_3591773']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_genes[\"antiSMASH\"] = df_genes[\"TranscriptId\"].apply(lambda x: dict_antismash[x])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check what type of mutations have occured in doxorubicin cluster"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This section deals with the genes in the doxorubicin cluster. Which genes are affected and with what kind of mutations? Besides that the genes are annotated with their gene code from a txt file."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select only the genes in the DXR cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# select only the genes of the doxorubicin cluster\n",
    "df_DXR = df_genes.loc[(df_genes[\"GeneId\"] > 24485) & (df_genes[\"GeneId\"] < 24680)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# For each gene, find which mutations affect it and collect in a dataframe\n",
    "df_old = df_ann[df_ann['ANN'].str.contains('CGZ69_RS24500')]\n",
    "\n",
    "for gene_id in df_DXR[\"TranscriptId\"]:\n",
    "    df_new = df_ann[df_ann['Gene_ID'].str.contains(gene_id)]\n",
    "    df_ann_DXR = pd.concat([df_old, df_new])\n",
    "    df_old = df_ann_DXR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove duplicates and remove mutations outside genes: modifiers\n",
    "df_ann_DXR = df_ann_DXR.sort_index().drop_duplicates()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Add gene names to DXR genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Extract gene names from txt file and create dictionary\n",
    "with open(fun.get_file_path(\"DXR_cluster_ANN.txt\"), 'r') as file:\n",
    "    x = file.read().splitlines()\n",
    "\n",
    "dict_DXR_annotation = {}\n",
    "\n",
    "for line in x:\n",
    "    row = line.split(\"\\t\")\n",
    "    dict_DXR_annotation[row[1]] = row[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make a list of genes ordered like the dataframe and replace current gene names\n",
    "genes_ordered = []\n",
    "\n",
    "for geneID in df_ann_DXR[\"Gene_ID\"]:\n",
    "    genes_ordered.append(dict_DXR_annotation[geneID])\n",
    "\n",
    "df_ann_DXR[\"Gene_Name\"] = genes_ordered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Extract usefull data from dataframe\n",
    "df_ann_DXR = df_ann_DXR[[\"Gene_ID\", \"Gene_Name\", \"Annotation\", \"Annotation_Impact\", \"HGVS.c\", \"CDS.pos / CDS.length\", \"HGVS.p\", \"AA.pos / AA.length\"]]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Find interesting genes in DXR cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of genes in the DXR cluster: 38\n",
      "Genes in the DXR cluster with a mutation in coding sequence: 9\n",
      "Genes in the DXR cluster with a high or moderate mutation in coding sequence: 5\n",
      "Total number of mutations in the DXR cluster: 15\n",
      "Mutations in the DXR cluster within a coding sequence: 11\n",
      "Mutations in the DXR cluster in a promoter sequence (300 nt upstream): 0\n"
     ]
    }
   ],
   "source": [
    "df_DXR_coding = df_DXR.loc[(df_genes[\"variants_impact_HIGH\"] > 0) | (df_genes[\"variants_impact_MODERATE\"] > 0) | (df_genes[\"variants_impact_LOW\"] > 0)]\n",
    "df_DXR_coding_high_mod = df_DXR.loc[(df_genes[\"variants_impact_HIGH\"] > 0) | (df_genes[\"variants_impact_MODERATE\"] > 0)]\n",
    "\n",
    "df_ann_DXR_coding = df_ann_DXR[df_ann_DXR[\"Annotation_Impact\"] != \"MODIFIER\"]\n",
    "df_ann_DXR_promoter = df_ann_DXR.loc[(df_ann_DXR[\"Annotation\"] == \"upstream_gene_variant\") & (df_ann[\"Distance\"] < 300)]\n",
    "\n",
    "print(\"Total number of genes in the DXR cluster:\", len(df_DXR))\n",
    "print(\"Genes in the DXR cluster with a mutation in coding sequence:\", len(df_DXR_coding))\n",
    "print(\"Genes in the DXR cluster with a high or moderate mutation in coding sequence:\", len(df_DXR_coding_high_mod))\n",
    "\n",
    "print(\"Total number of mutations in the DXR cluster:\", len(df_ann_DXR))\n",
    "print(\"Mutations in the DXR cluster within a coding sequence:\", len(df_ann_DXR_coding))\n",
    "print(\"Mutations in the DXR cluster in a promoter sequence (300 nt upstream):\", len(df_ann_DXR_promoter))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Annotation_Impact</th>\n",
       "      <th>HGVS.c</th>\n",
       "      <th>CDS.pos / CDS.length</th>\n",
       "      <th>HGVS.p</th>\n",
       "      <th>AA.pos / AA.length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1478</th>\n",
       "      <td>CGZ69_RS24490</td>\n",
       "      <td>drrC</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>c.1631C&gt;T</td>\n",
       "      <td>1631/2295</td>\n",
       "      <td>p.Pro544Leu</td>\n",
       "      <td>544/764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1479</th>\n",
       "      <td>CGZ69_RS24490</td>\n",
       "      <td>drrC</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>c.1670C&gt;T</td>\n",
       "      <td>1670/2295</td>\n",
       "      <td>p.Thr557Ile</td>\n",
       "      <td>557/764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1480</th>\n",
       "      <td>CGZ69_RS24500</td>\n",
       "      <td>dnrS</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>c.67G&gt;A</td>\n",
       "      <td>67/1296</td>\n",
       "      <td>p.Ala23Thr</td>\n",
       "      <td>23/431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1481</th>\n",
       "      <td>CGZ69_RS24515</td>\n",
       "      <td>dnrK</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>c.19G&gt;A</td>\n",
       "      <td>19/1071</td>\n",
       "      <td>p.Val7Ile</td>\n",
       "      <td>7/356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1487</th>\n",
       "      <td>CGZ69_RS24570</td>\n",
       "      <td>dnrH</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>c.1100G&gt;A</td>\n",
       "      <td>1100/1332</td>\n",
       "      <td>p.Gly367Asp</td>\n",
       "      <td>367/443</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1491</th>\n",
       "      <td>CGZ69_RS24645</td>\n",
       "      <td>drrD</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>c.1068delG</td>\n",
       "      <td>1068/1191</td>\n",
       "      <td>p.Gly358fs</td>\n",
       "      <td>356/396</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Gene_ID Gene_Name          Annotation Annotation_Impact  \\\n",
       "1478  CGZ69_RS24490      drrC    missense_variant          MODERATE   \n",
       "1479  CGZ69_RS24490      drrC    missense_variant          MODERATE   \n",
       "1480  CGZ69_RS24500      dnrS    missense_variant          MODERATE   \n",
       "1481  CGZ69_RS24515      dnrK    missense_variant          MODERATE   \n",
       "1487  CGZ69_RS24570      dnrH    missense_variant          MODERATE   \n",
       "1491  CGZ69_RS24645      drrD  frameshift_variant              HIGH   \n",
       "\n",
       "          HGVS.c CDS.pos / CDS.length       HGVS.p AA.pos / AA.length  \n",
       "1478   c.1631C>T            1631/2295  p.Pro544Leu            544/764  \n",
       "1479   c.1670C>T            1670/2295  p.Thr557Ile            557/764  \n",
       "1480     c.67G>A              67/1296   p.Ala23Thr             23/431  \n",
       "1481     c.19G>A              19/1071    p.Val7Ile              7/356  \n",
       "1487   c.1100G>A            1100/1332  p.Gly367Asp            367/443  \n",
       "1491  c.1068delG            1068/1191   p.Gly358fs            356/396  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Genes in the DXR cluster with a high or moderate mutation in coding sequence\n",
    "df_ann_DXR[(df_ann_DXR[\"Annotation_Impact\"] == \"HIGH\") | (df_ann_DXR[\"Annotation_Impact\"] == \"MODERATE\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Annotation_Impact</th>\n",
       "      <th>HGVS.c</th>\n",
       "      <th>CDS.pos / CDS.length</th>\n",
       "      <th>HGVS.p</th>\n",
       "      <th>AA.pos / AA.length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1484</th>\n",
       "      <td>CGZ69_RS24535</td>\n",
       "      <td>dpsC</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>c.-4950T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1485</th>\n",
       "      <td>CGZ69_RS24535</td>\n",
       "      <td>dpsC</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>c.-4954T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1489</th>\n",
       "      <td>CGZ69_RS24565</td>\n",
       "      <td>dnrE</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>c.-3908G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1492</th>\n",
       "      <td>CGZ69_RS24650</td>\n",
       "      <td>drrB</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>c.-4128_-4118delCCCTCCCGGTA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Gene_ID Gene_Name             Annotation Annotation_Impact  \\\n",
       "1484  CGZ69_RS24535      dpsC  upstream_gene_variant          MODIFIER   \n",
       "1485  CGZ69_RS24535      dpsC  upstream_gene_variant          MODIFIER   \n",
       "1489  CGZ69_RS24565      dnrE  upstream_gene_variant          MODIFIER   \n",
       "1492  CGZ69_RS24650      drrB  upstream_gene_variant          MODIFIER   \n",
       "\n",
       "                           HGVS.c CDS.pos / CDS.length HGVS.p  \\\n",
       "1484                   c.-4950T>G                               \n",
       "1485                   c.-4954T>G                               \n",
       "1489                   c.-3908G>A                               \n",
       "1492  c.-4128_-4118delCCCTCCCGGTA                               \n",
       "\n",
       "     AA.pos / AA.length  \n",
       "1484                     \n",
       "1485                     \n",
       "1489                     \n",
       "1492                     "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Genes in the DXR cluster oustide the coding sequence\n",
    "df_ann_DXR[df_ann_DXR[\"Annotation_Impact\"] == \"MODIFIER\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check mutations in other genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### How many genes are affected and by what type of mutations?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of genes: 7129\n",
      "Genes affected by a mutation: 134 (79 % of all genes)\n",
      "Genes with mutation inside coding region only: 134 (1.9 % of all genes)\n",
      "Genes with mutation outside coding region only: 4583 (64 % of all genes)\n",
      "Genes with mutation inside and outside coding region: 908 (13 % of all genes)\n"
     ]
    }
   ],
   "source": [
    "# Which genes are affected by the mutations?\n",
    "total_genes = 7129\n",
    "\n",
    "df_genes_affected = df_genes.loc[(df_genes[\"variants_impact_HIGH\"] > 0) | \n",
    "                                 (df_genes[\"variants_impact_MODERATE\"] > 0) | \n",
    "                                 (df_genes[\"variants_impact_LOW\"] > 0) |\n",
    "                                 (df_genes[\"variants_impact_MODIFIER\"] > 0)]\n",
    "\n",
    "df_genes_coding = df_genes[(df_genes[\"variants_impact_MODIFIER\"] == 0)]\n",
    "\n",
    "df_genes_surrounding = df_genes.loc[(df_genes[\"variants_impact_HIGH\"] == 0) & \n",
    "                                    (df_genes[\"variants_impact_MODERATE\"] == 0) & \n",
    "                                    (df_genes[\"variants_impact_LOW\"] == 0)]\n",
    "\n",
    "df_genes_cod_sur = df_genes.loc[((df_genes[\"variants_impact_HIGH\"] > 0) | (df_genes[\"variants_impact_MODERATE\"] > 0) | (df_genes[\"variants_impact_LOW\"] > 0)) & \n",
    "                                (df_genes[\"variants_impact_MODIFIER\"] > 0)]\n",
    "\n",
    "print((\"Total number of genes: {}\").format(total_genes))\n",
    "print((\"Genes affected by a mutation: {} ({:.2g} % of all genes)\").format(len(df_genes_coding), len(df_genes_affected)/total_genes*100))\n",
    "print((\"Genes with mutation inside coding region only: {} ({:.2g} % of all genes)\").format(len(df_genes_coding), len(df_genes_coding)/total_genes*100))\n",
    "print((\"Genes with mutation outside coding region only: {} ({:.2g} % of all genes)\").format(len(df_genes_surrounding), len(df_genes_surrounding)/total_genes*100))\n",
    "print((\"Genes with mutation inside and outside coding region: {} ({:.2g} % of all genes)\").format(len(df_genes_cod_sur), len(df_genes_cod_sur)/total_genes*100))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of mutations: 2221\n",
      "Mutations with a HIGH impact: 370 (17 % of all mutations)\n",
      "   of which 25 affecting stop codons and 347 affecting frameshifts.\n",
      "Mutations with a MODERATE impact: 859 (39 % of all mutations)\n",
      "   of which 2 cause a disruptive and 3 a conservative inframe deletion, and 854 an amino acid change.\n",
      "Mutations with a LOW impact: 432 (19 % of all mutations)\n",
      "Mutations in a promoter region: 50 (2.3 % of all mutations)\n"
     ]
    }
   ],
   "source": [
    "# What type of mutations have occurred?\n",
    "total_mutations = 2221\n",
    "\n",
    "df_ann_high = df_ann[df_ann[\"Annotation_Impact\"] == \"HIGH\"]\n",
    "df_ann_high_stop = df_ann_high[df_ann_high[\"Annotation\"].str.contains(\"stop\")]\n",
    "df_ann_high_frameshift = df_ann_high[df_ann_high[\"Annotation\"].str.contains(\"frameshift\")]\n",
    "df_ann_moderate = df_ann[df_ann[\"Annotation_Impact\"] == \"MODERATE\"]\n",
    "df_ann_moderate_disr = df_ann_moderate[df_ann_moderate[\"Annotation\"].str.contains(\"disruptive\")]\n",
    "df_ann_moderate_cons = df_ann_moderate[df_ann_moderate[\"Annotation\"].str.contains(\"conservative\")]\n",
    "df_ann_moderate_amino = df_ann_moderate[df_ann_moderate[\"Annotation\"].str.contains(\"missense\")]\n",
    "df_ann_low = df_ann[df_ann[\"Annotation_Impact\"] == \"LOW\"]\n",
    "df_ann_promoter = df_ann[(df_ann[\"Annotation\"] == \"upstream_gene_variant\") & (df_ann[\"Distance\"] < 300)]\n",
    "\n",
    "print((\"Total number of mutations: {}\").format(total_mutations))\n",
    "print((\"Mutations with a HIGH impact: {} ({:.2g} % of all mutations)\").format(len(df_ann_high), len(df_ann_high)/total_mutations*100))\n",
    "print((\"   of which {} affecting stop codons and {} affecting frameshifts.\").format(len(df_ann_high_stop), len(df_ann_high_frameshift)))\n",
    "print((\"Mutations with a MODERATE impact: {} ({:.2g} % of all mutations)\").format(len(df_ann_moderate), len(df_ann_moderate)/total_mutations*100))\n",
    "print((\"   of which {} cause a disruptive and {} a conservative inframe deletion, and {} an amino acid change.\")\n",
    "      .format(len(df_ann_moderate_disr), len(df_ann_moderate_cons), len(df_ann_moderate_amino)))\n",
    "print((\"Mutations with a LOW impact: {} ({:.2g} % of all mutations)\").format(len(df_ann_low), len(df_ann_low)/total_mutations*100))\n",
    "print((\"Mutations in a promoter region: {} ({:.2g} % of all mutations)\").format(len(df_ann_promoter), len(df_ann_promoter)/total_mutations*100))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Which mutations are interesting?\n",
    "df_ann_interesting = df_ann[(df_ann[\"Annotation_Impact\"] == \"HIGH\") | \n",
    "                            (df_ann[\"Annotation_Impact\"] == \"MODERATE\") | \n",
    "                            ((df_ann[\"Annotation_Impact\"] == \"MODIFIER\") & (df_ann[\"Distance\"] < 300))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Which interesting mutations have a gene name?\n",
    "gene_int_list = df_ann_interesting[~df_ann_interesting[\"Gene_Name\"].str.contains(\"CGZ\")][\"Gene_Name\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add a column to df_genes with the genes that have a mutation in the promoter region.\n",
    "df_genes[\"variants_impact_promoter\"] = 0\n",
    "\n",
    "for gene_code in df_ann_promoter[\"Gene_ID\"]:\n",
    "    df_genes.loc[df_genes['TranscriptId'] == gene_code,\"variants_impact_promoter\"] += 1 \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_genes.loc[df_genes['#GeneName'].str.contains('CGZ69'), '#GeneName'] = ''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Which genes have interesting mutations?\n",
    "df_genes_interesting = df_genes[(df_genes[\"variants_impact_HIGH\"] > 0) | \n",
    "                                (df_genes[\"variants_impact_MODERATE\"] > 0) | \n",
    "                                (df_genes[\"variants_impact_promoter\"] > 0)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#GeneName</th>\n",
       "      <th>GeneId</th>\n",
       "      <th>TranscriptId</th>\n",
       "      <th>BioType</th>\n",
       "      <th>variants_impact_HIGH</th>\n",
       "      <th>variants_impact_LOW</th>\n",
       "      <th>variants_impact_MODERATE</th>\n",
       "      <th>variants_impact_MODIFIER</th>\n",
       "      <th>variants_effect_conservative_inframe_deletion</th>\n",
       "      <th>variants_effect_conservative_inframe_insertion</th>\n",
       "      <th>variants_effect_disruptive_inframe_deletion</th>\n",
       "      <th>variants_effect_downstream_gene_variant</th>\n",
       "      <th>variants_effect_frameshift_variant</th>\n",
       "      <th>variants_effect_missense_variant</th>\n",
       "      <th>variants_effect_splice_region_variant</th>\n",
       "      <th>variants_effect_stop_gained</th>\n",
       "      <th>variants_effect_stop_lost</th>\n",
       "      <th>variants_effect_synonymous_variant</th>\n",
       "      <th>variants_effect_upstream_gene_variant</th>\n",
       "      <th>SCO</th>\n",
       "      <th>antiSMASH</th>\n",
       "      <th>variants_impact_promoter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td></td>\n",
       "      <td>225</td>\n",
       "      <td>CGZ69_RS00225</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>pseudo_gene</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td></td>\n",
       "      <td>245</td>\n",
       "      <td>CGZ69_RS00245</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SCO6220</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td></td>\n",
       "      <td>470</td>\n",
       "      <td>CGZ69_RS00470</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NotFound</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td></td>\n",
       "      <td>490</td>\n",
       "      <td>CGZ69_RS00490</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SCO6219</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td></td>\n",
       "      <td>535</td>\n",
       "      <td>CGZ69_RS00535</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NotFound</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "0  #GeneName  GeneId   TranscriptId         BioType  variants_impact_HIGH  \\\n",
       "12               225  CGZ69_RS00225  protein_coding                     0   \n",
       "16               245  CGZ69_RS00245  protein_coding                     0   \n",
       "36               470  CGZ69_RS00470  protein_coding                     0   \n",
       "40               490  CGZ69_RS00490  protein_coding                     0   \n",
       "47               535  CGZ69_RS00535  protein_coding                     0   \n",
       "\n",
       "0   variants_impact_LOW  variants_impact_MODERATE  variants_impact_MODIFIER  \\\n",
       "12                    0                         1                         0   \n",
       "16                    0                         1                         1   \n",
       "36                    0                         1                         0   \n",
       "40                    0                         1                         1   \n",
       "47                    0                         1                         1   \n",
       "\n",
       "0  variants_effect_conservative_inframe_deletion  \\\n",
       "12                                             0   \n",
       "16                                             0   \n",
       "36                                             0   \n",
       "40                                             0   \n",
       "47                                             0   \n",
       "\n",
       "0  variants_effect_conservative_inframe_insertion  \\\n",
       "12                                              0   \n",
       "16                                              0   \n",
       "36                                              0   \n",
       "40                                              0   \n",
       "47                                              0   \n",
       "\n",
       "0  variants_effect_disruptive_inframe_deletion  \\\n",
       "12                                           0   \n",
       "16                                           0   \n",
       "36                                           0   \n",
       "40                                           0   \n",
       "47                                           0   \n",
       "\n",
       "0  variants_effect_downstream_gene_variant variants_effect_frameshift_variant  \\\n",
       "12                                       0                                  0   \n",
       "16                                       1                                  0   \n",
       "36                                       0                                  0   \n",
       "40                                       1                                  0   \n",
       "47                                       0                                  0   \n",
       "\n",
       "0  variants_effect_missense_variant variants_effect_splice_region_variant  \\\n",
       "12                                1                                     0   \n",
       "16                                1                                     0   \n",
       "36                                1                                     0   \n",
       "40                                1                                     0   \n",
       "47                                1                                     0   \n",
       "\n",
       "0  variants_effect_stop_gained variants_effect_stop_lost  \\\n",
       "12                           0                         0   \n",
       "16                           0                         0   \n",
       "36                           0                         0   \n",
       "40                           0                         0   \n",
       "47                           0                         0   \n",
       "\n",
       "0  variants_effect_synonymous_variant variants_effect_upstream_gene_variant  \\\n",
       "12                                  0                                     0   \n",
       "16                                  0                                     0   \n",
       "36                                  0                                     0   \n",
       "40                                  0                                     0   \n",
       "47                                  0                                     1   \n",
       "\n",
       "0           SCO antiSMASH  variants_impact_promoter  \n",
       "12  pseudo_gene                                   0  \n",
       "16      SCO6220                                   0  \n",
       "36     NotFound                                   0  \n",
       "40      SCO6219                                   0  \n",
       "47     NotFound                                   0  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_genes_interesting.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_SCO_out = df_genes_interesting[[\"TranscriptId\", \"#GeneName\", \"SCO\", \"variants_impact_HIGH\", \"variants_impact_MODERATE\", \"variants_impact_promoter\", \"antiSMASH\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TranscriptId</th>\n",
       "      <th>#GeneName</th>\n",
       "      <th>SCO</th>\n",
       "      <th>variants_impact_HIGH</th>\n",
       "      <th>variants_impact_MODERATE</th>\n",
       "      <th>variants_impact_promoter</th>\n",
       "      <th>antiSMASH</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CGZ69_RS00245</td>\n",
       "      <td></td>\n",
       "      <td>SCO6220</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>CGZ69_RS00490</td>\n",
       "      <td></td>\n",
       "      <td>SCO6219</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>CGZ69_RS00805</td>\n",
       "      <td></td>\n",
       "      <td>SCO1569</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>CGZ69_RS01200</td>\n",
       "      <td></td>\n",
       "      <td>SCO3444</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>CGZ69_RS01340</td>\n",
       "      <td></td>\n",
       "      <td>SCO6479</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5585</th>\n",
       "      <td>CGZ69_RS16085</td>\n",
       "      <td>thpR</td>\n",
       "      <td>SCO4338</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5605</th>\n",
       "      <td>CGZ69_RS21230</td>\n",
       "      <td>tuf</td>\n",
       "      <td>SCO4662</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5608</th>\n",
       "      <td>CGZ69_RS16535</td>\n",
       "      <td>ugpC</td>\n",
       "      <td>SCO4240</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5614</th>\n",
       "      <td>CGZ69_RS08535</td>\n",
       "      <td>uvrB</td>\n",
       "      <td>SCO1966</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5624</th>\n",
       "      <td>CGZ69_RS29290</td>\n",
       "      <td>zapE</td>\n",
       "      <td>SCO6057</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>472 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "0      TranscriptId #GeneName      SCO  variants_impact_HIGH  \\\n",
       "16    CGZ69_RS00245            SCO6220                     0   \n",
       "40    CGZ69_RS00490            SCO6219                     0   \n",
       "80    CGZ69_RS00805            SCO1569                     0   \n",
       "141   CGZ69_RS01200            SCO3444                     0   \n",
       "164   CGZ69_RS01340            SCO6479                     0   \n",
       "...             ...       ...      ...                   ...   \n",
       "5585  CGZ69_RS16085      thpR  SCO4338                     0   \n",
       "5605  CGZ69_RS21230       tuf  SCO4662                     0   \n",
       "5608  CGZ69_RS16535      ugpC  SCO4240                     0   \n",
       "5614  CGZ69_RS08535      uvrB  SCO1966                     0   \n",
       "5624  CGZ69_RS29290      zapE  SCO6057                     0   \n",
       "\n",
       "0     variants_impact_MODERATE  variants_impact_promoter antiSMASH  \n",
       "16                           1                         0            \n",
       "40                           1                         0            \n",
       "80                           0                         1            \n",
       "141                          1                         0            \n",
       "164                          1                         0            \n",
       "...                        ...                       ...       ...  \n",
       "5585                         1                         0            \n",
       "5605                         1                         0            \n",
       "5608                         2                         0            \n",
       "5614                         1                         0            \n",
       "5624                         1                         0            \n",
       "\n",
       "[472 rows x 7 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_SCO_out[(df_SCO_out[\"SCO\"] != \"pseudo_gene\") & (df_SCO_out[\"SCO\"] != \"NotFound\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_SCO_out.to_csv(\"df_SCO_out.csv\", index=False, sep = ';')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check one specific gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check mutations in one specific gene\n",
    "this_gene = \"CGZ69_RS24515\"\n",
    "df_ann_this_gene = df_ann[df_ann['ANN'].str.contains(this_gene)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>ANN</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Annotation_Impact</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Feature_Type</th>\n",
       "      <th>Feature_ID</th>\n",
       "      <th>Transcript_BioType</th>\n",
       "      <th>Rank</th>\n",
       "      <th>HGVS.c</th>\n",
       "      <th>HGVS.p</th>\n",
       "      <th>cDNA.pos / cDNA.length</th>\n",
       "      <th>CDS.pos / CDS.length</th>\n",
       "      <th>AA.pos / AA.length</th>\n",
       "      <th>Distance</th>\n",
       "      <th>ERRORS / WARNINGS / INFO</th>\n",
       "      <th>SCO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1480</th>\n",
       "      <td>5283337</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>225.0</td>\n",
       "      <td>T|missense_variant|MODERATE|CGZ69_RS24500|CGZ6...</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>CGZ69_RS24500</td>\n",
       "      <td>CGZ69_RS24500</td>\n",
       "      <td>transcript</td>\n",
       "      <td>CGZ69_RS24500</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td>c.67G&gt;A</td>\n",
       "      <td>p.Ala23Thr</td>\n",
       "      <td>67/1296</td>\n",
       "      <td>67/1296</td>\n",
       "      <td>23/431</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NotFound</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1481</th>\n",
       "      <td>5286729</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>225.0</td>\n",
       "      <td>T|missense_variant|MODERATE|CGZ69_RS24515|CGZ6...</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>CGZ69_RS24515</td>\n",
       "      <td>CGZ69_RS24515</td>\n",
       "      <td>transcript</td>\n",
       "      <td>CGZ69_RS24515</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td>c.19G&gt;A</td>\n",
       "      <td>p.Val7Ile</td>\n",
       "      <td>19/1071</td>\n",
       "      <td>19/1071</td>\n",
       "      <td>7/356</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>SCO7452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1482</th>\n",
       "      <td>5288942</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>225.0</td>\n",
       "      <td>T|synonymous_variant|LOW|CGZ69_RS24530|CGZ69_R...</td>\n",
       "      <td>T</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>CGZ69_RS24530</td>\n",
       "      <td>CGZ69_RS24530</td>\n",
       "      <td>transcript</td>\n",
       "      <td>CGZ69_RS24530</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td>c.327G&gt;A</td>\n",
       "      <td>p.Ala109Ala</td>\n",
       "      <td>327/1017</td>\n",
       "      <td>327/1017</td>\n",
       "      <td>109/338</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>SCO6275</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          POS REF ALT   QUAL  \\\n",
       "1480  5283337   C   T  225.0   \n",
       "1481  5286729   C   T  225.0   \n",
       "1482  5288942   C   T  225.0   \n",
       "\n",
       "                                                    ANN Allele  \\\n",
       "1480  T|missense_variant|MODERATE|CGZ69_RS24500|CGZ6...      T   \n",
       "1481  T|missense_variant|MODERATE|CGZ69_RS24515|CGZ6...      T   \n",
       "1482  T|synonymous_variant|LOW|CGZ69_RS24530|CGZ69_R...      T   \n",
       "\n",
       "              Annotation Annotation_Impact      Gene_Name        Gene_ID  \\\n",
       "1480    missense_variant          MODERATE  CGZ69_RS24500  CGZ69_RS24500   \n",
       "1481    missense_variant          MODERATE  CGZ69_RS24515  CGZ69_RS24515   \n",
       "1482  synonymous_variant               LOW  CGZ69_RS24530  CGZ69_RS24530   \n",
       "\n",
       "     Feature_Type     Feature_ID Transcript_BioType Rank    HGVS.c  \\\n",
       "1480   transcript  CGZ69_RS24500     protein_coding  1/1   c.67G>A   \n",
       "1481   transcript  CGZ69_RS24515     protein_coding  1/1   c.19G>A   \n",
       "1482   transcript  CGZ69_RS24530     protein_coding  1/1  c.327G>A   \n",
       "\n",
       "           HGVS.p cDNA.pos / cDNA.length CDS.pos / CDS.length  \\\n",
       "1480   p.Ala23Thr                67/1296              67/1296   \n",
       "1481    p.Val7Ile                19/1071              19/1071   \n",
       "1482  p.Ala109Ala               327/1017             327/1017   \n",
       "\n",
       "     AA.pos / AA.length  Distance ERRORS / WARNINGS / INFO       SCO  \n",
       "1480             23/431       NaN                           NotFound  \n",
       "1481              7/356       NaN                            SCO7452  \n",
       "1482            109/338       NaN                            SCO6275  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ann_this_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
